test/over_react/component_declaration/component_base_test.dart

```diff
@@ -996,6 +1091,40 @@ class TestComponentComponent extends UiComponent<TestComponentProps> {
   }
 }
 
+class TestComponent2Props extends UiProps {
```

Comment:
Is there a reason we wouldn't just create a new component declaration in a separate file for this stuff instead of manually creating classes that represent what the builder would generate?

Seems brittle to me to be hard-coding class members that are generated IRL.

Reply:
I agree it is less than ideal. @kealjones-wk and I actually asked the same thing originally - we may have had other thoughts that I can't recall, but the big one is that the way it's set up now the test can easily specify the consumed props through the constructor. That way someone can actually write tests with the consumed props changing and you can validate everything works with pretty minimal code. To do the same thing through generated code seemed more complex. 

With all that being the case, would it be better to put the effort in to create the generated componentry?

Reply:
Looks like we are already doing this in other places, and its not important enough to break pattern for now.